src/device-state/current-state.ts

```diff
@@ -177,7 +177,10 @@ const reportCurrentState = (): null => {
 		reportPending = true;
 		try {
 			const currentDeviceState = await deviceState.getStatus();
-			const info = await sysInfo.getSysInfoToReport();
+			const info = await sysInfo.getSysInfoToReport(
+				await config.get('reportHardwareMetrics'),
+			);
```

Comment:
I think adding the toggle to the report generate function adds necessary complexity to the signature when we can do:

```
	const info = (await config.get('reportHardwareMetrics'))
		? await sysInfo.getSysInfoToReport()
		: {};
```

My other concern with getting the config every time is if it adds another IO we can prevent. Especially since this happens every 10 seconds.. What if we do something like this to cache the value and update the cache as it changes:

```
let reportHardwareMetrics: boolean;

async function shouldReportMetrics() {
	if (!reportHardwareMetrics){
		reportHardwareMetrics =	await config.get('reportHardwareMetrics')
	}
	return reportHardwareMetrics
}

config.on('change', (changed) => {
	if (changed.reportHardwareMetrics != null) {
		// this value has changed
		reportHardwareMetrics = changed.reportHardwareMetrics
	}
});
```
then get info like:
```
	const info = (await shouldReportMetrics())
		? await sysInfo.getSysInfoToReport()
		: {};
```

Reply:
@20k-ultra This makes sense. Realistically SQLite3 should be able to handle many more queries per second than Supervisor is throwing at it, even with the reportCurrentState function making a SELECT query every 10 seconds, but we should also consider memory limitations on the device. While writing this PR I was thinking of whether to cache the reportHardware config value, so took a look at existing code to see if there were any existing examples. I'm down to make the change here, but if so, the caching change should also be made in the [actual report function](https://github.com/balena-os/balena-supervisor/blob/master/src/device-state/current-state.ts#L134-L140), which also fires every 10 seconds max.

Reply:
In fact we query for some of the same values [twice](https://github.com/balena-os/balena-supervisor/blob/master/src/device-state/current-state.ts#L71-L76) every 10 seconds :|

Reply:
I think this makes it clear that we shouldn't implement the config retrieve cache in each module but just in the config component itself. Feel free to ignore the caching comment and we can do it in https://github.com/balena-os/balena-supervisor/issues/1667. 

Do you agree with not asking sysInfo to generate hte report if the feature is disabled though? I think you do.

Reply:
For sure, I wrote a comment in #1667 and we can look more into in the future.

> Do you agree with not asking sysInfo to generate hte report if the feature is disabled though? I think you do.

I agree and I've already made the changes locally, just haven't updated the remote.